src/space.cpp

```diff
@@ -488,18 +488,22 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp, bool comp
       }
 
       // Issue #1005
-      if (!((first->type == CT_TYPE) &&
-            (first->prev != nullptr) &&
-            (first->prev->type == CT_FRIEND)))
-      {
-         if ((first->type == CT_WORD) ||
-             (first->type == CT_PAREN_CLOSE) ||
-             (first->type == CT_TYPE) ||
-             CharTable::IsKw1(first->str[0]))
-         {
-            log_rule("sp_before_dc");
-            return(cpd.settings[UO_sp_before_dc].a);
-         }
+      /* '::' at the start of an identifier is not member access, but global scope operator.
+       * Detect if previous chunk is a type and previous-previous is "friend"
+       */
+      if ((first->type == CT_TYPE) &&
+          (first->prev != nullptr) &&
+          (first->prev->type == CT_FRIEND))
+      {
+         log_rule("FORCE");
+         return(AV_FORCE);
```

Comment:
I don't think that this is 100% as it should be. 
Sorry, I should have mentioned this earlier, the problem is that in only in certain cases (the one you are checking here) combined with `UO_sp_before_dc = remove` that that remove needs to be changed to force. In every other case the settings should keep on working, including the case when `UO_sp_before_dc = add`. 
Certainly there will be someone out there who likes to keep his `n > 1` spaces.